rust/tools/qltest.sh

```diff
@@ -7,26 +7,20 @@ export RUST_BACKTRACE=full
 QLTEST_LOG="$CODEQL_EXTRACTOR_RUST_LOG_DIR"/qltest.log
 
 EXTRACTOR="$CODEQL_EXTRACTOR_RUST_ROOT/tools/$CODEQL_PLATFORM/extractor"
+echo > lib.rs
 for src in *.rs; do
-  echo -e "[workspace]\n\n[package]\nname = \"test\"\nversion=\"0.0.1\"\n[lib]\npath=\"$src\"\n" > Cargo.toml
-  env=()
-  opts=("$src")
-  opts+=($(sed -n '1 s=//codeql-extractor-options:==p' $src))
-  expected_status=$(sed -n 's=//codeql-extractor-expected-status:[[:space:]]*==p' $src)
-  expected_status=${expected_status:-0}
-  env+=($(sed -n '1 s=//codeql-extractor-env:==p' $src))
-  echo >> $QLTEST_LOG
-  echo "env ${env[@]} $EXTRACTOR ${opts[@]}" >> "$QLTEST_LOG"
-  env "${env[@]}" "$EXTRACTOR" "${opts[@]}" >> $QLTEST_LOG 2>&1
-  actual_status=$?
-  if [[ $actual_status != $expected_status ]]; then
-    FAILED=1
-  fi
+  echo "mod ${src%.rs};" >> lib.rs
 done
-
-rm -rf Cargo.*
-
-if [ -n "$FAILED" ]; then
+cat > Cargo.toml << EOF
```

Comment:
I don't really like writing random stuff to the source directory. Not sure it can be avoided. Have you tried generating a rust analyzer project json file instead of a Cargo file?

Reply:
No, I think that is a great idea, but probably rather as a fallback in the extractor itself. I would rather not try that out in the context of this PR and defer to future work.

Reply:
Makes sense, generating JSON in a bash script is a bit of a hassle. The extractor already picks up those json files if present. The "find_project_manifests" function looks for both types of files.

Reply:
yep! I think we might still want to implement a mode (either as an opt-in or a fallback when finding a manifest fails), where we just give files to the extractor and we do our best with those. rust-analyzer seems to have functionality for that, but we need to explore it. Then that mode might become the appropriate one for QL tests